testng-core/src/test/java/test/timeout/github2672/TimeoutStacktraceTest.java

```diff
@@ -0,0 +1,42 @@
+package test.timeout.github2672;
+
+import static org.testng.Assert.assertTrue;
+
+import java.util.Arrays;
+import org.testng.ITestListener;
+import org.testng.ITestResult;
+import org.testng.TestNG;
+import org.testng.annotations.Test;
+import org.testng.internal.thread.ThreadTimeoutException;
+
+public class TimeoutStacktraceTest {
```

Comment:
Please include a reference to this test class into `src/test/resources/testng.xml` so that it gets included in every execution.